lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -1904,6 +1904,32 @@ namespace Js
         return true;
     }
 
+#ifdef ENABLE_JS_BUILTINS
+    void EnsureBuiltInEngineIsReady(JsBuiltInFile file, ScriptContext* scriptContext)
+    {
+        if (scriptContext->IsJsBuiltInEnabled())
+        {
+            if (scriptContext->VerifyAlive())  // Can't initialize if scriptContext closed, will need to run script
+            {
+                Assert(engineInterfaceObject != nullptr);
+                JsBuiltInEngineInterfaceExtensionObject* builtInExtension =
+                    static_cast<JsBuiltInEngineInterfaceExtensionObject*>(scriptContext->GetLibrary()->GetEngineInterfaceObject()->GetEngineExtension(EngineInterfaceExtensionKind_JsBuiltIn));
+                builtInExtension->InjectJsBuiltInLibraryCode(scriptContext, file);
+            }
+        }
+    }
+    
+    void JavascriptLibrary::EnsureArrayBuiltInsAreReady()
+    {
+        EnsureBuiltInEngineIsReady(JsBuiltInFile::Array_prototype, scriptContext);
+    }
+
+    void JavascriptLibrary::EnsureMathBuiltInsAreReady()
+    {
+        EnsureBuiltInEngineIsReady(JsBuiltInFile::Math_object, scriptContext);
+    }
+#endif
```

Comment:
New methods for loading JsBuiltins - had to move up the file from the old ones to be above all calls to them as could not put the new EnsureBuiltInEngineIsReady in the header, the header does not have access to the JsBuiltInFile type due to the order of header includes.